src/core/ext/transport/chttp2/transport/decode_huff.h

```diff
@@ -0,0 +1,1018 @@
+    const int consumed = op & 7;
+    buffer_len_ -= consumed;
+    const auto emit_ofs = op >> 4;
+    switch ((op >> 3) & 1) {
+      case 1: {
+        DecodeStep13();
+        break;
+      }
+      case 0: {
+        sink_(GetEmit26(index, emit_ofs + 0));
+        break;
+      }
+    }
+  }
+  bool RefillTo5() {
+    switch (buffer_len_) {
+      case 0:
+      case 1:
+      case 2:
+      case 3:
+      case 4: {
+        return Read1();
+      }
+    }
+    return true;
+  }
+  void Done13() {
+    done_ = true;
+    switch (buffer_len_) {
+      case 4: {
+        const auto index = buffer_ & 15;
+        const auto op = GetOp27(index);
+        switch (op & 3) {
+          case 1: {
+            ok_ = false;
+            break;
+          }
+          case 2: {
+            sink_(GetEmit27(index, op >> 2));
+            break;
+          }
+        }
+        return;
+      }
+      case 1:
+      case 2:
+      case 3: {
+        ok_ = (buffer_ & ((1 << buffer_len_) - 1)) == (1 << buffer_len_) - 1;
+        return;
+      }
+      case 0: {
+        return;
+      }
+    }
+  }
+  void DecodeStep13() {
+    if (!RefillTo2()) {
+      Done14();
+      return;
+    }
+    const auto index = (buffer_ >> (buffer_len_ - 2)) & 0x3;
+    const auto op = GetOp28(index);
+    const int consumed = op & 3;
+    buffer_len_ -= consumed;
+    const auto emit_ofs = op >> 3;
+    switch ((op >> 2) & 1) {
+      case 1: {
+        begin_ = end_;
+        buffer_len_ = 0;
+        break;
+      }
+      case 0: {
+        sink_(GetEmit28(index, emit_ofs + 0));
+        break;
+      }
+    }
+  }
+  void Done14() {
+    done_ = true;
+    switch (buffer_len_) {
+      case 1: {
+        ok_ = (buffer_ & ((1 << buffer_len_) - 1)) == (1 << buffer_len_) - 1;
+        return;
+      }
+      case 0: {
+        return;
+      }
+    }
+  }
+  void DecodeStep14() {
+    if (!RefillTo4()) {
+      Done15();
+      return;
+    }
+    const auto index = (buffer_ >> (buffer_len_ - 4)) & 0xf;
+    const auto op = GetOp29(index);
+    const int consumed = op & 7;
+    buffer_len_ -= consumed;
+    const auto emit_ofs = op >> 3;
+    sink_(GetEmit29(index, emit_ofs + 0));
+  }
+  void Done15() {
+    done_ = true;
+    switch (buffer_len_) {
+      case 3: {
+        const auto index = buffer_ & 7;
+        const auto op = GetOp30(index);
+        switch (op & 3) {
+          case 1: {
+            ok_ = false;
+            break;
+          }
+          case 2: {
+            sink_(GetEmit30(index, op >> 2));
+            break;
+          }
+        }
+        return;
+      }
+      case 0:
+      case 1:
+      case 2: {
+        ok_ = false;
+        return;
+      }
+    }
+  }
+  F sink_;
+  const uint8_t* begin_;
+  const uint8_t* const end_;
+  uint64_t buffer_ = 0;
+  int buffer_len_ = 0;
+  bool ok_ = true;
+  bool done_ = false;
+};
+}  // namespace grpc_core
+#endif  // GRPC_CORE_EXT_TRANSPORT_CHTTP2_TRANSPORT_DECODE_HUFF_H
```

Comment:
please fix the no new line at end of file

Reply:
done